File: pg-index-health-spring-boot-demo/src/main/java/io/github/mfvanek/pg/index/health/demo/config/MigrationGeneratorConfig.java
/*
 * Copyright (c) 2019-2024. Ivan Vakhrushev and others.
 * https://github.com/mfvanek/pg-index-health-demo
 *
 * Licensed under the Apache License 2.0
 */

package io.github.mfvanek.pg.index.health.demo.config;

import io.github.mfvanek.pg.connection.HighAvailabilityPgConnection;
import io.github.mfvanek.pg.generator.DbMigrationGenerator;
import io.github.mfvanek.pg.generator.ForeignKeyMigrationGenerator;
import io.github.mfvanek.pg.generator.GeneratingOptions;
import io.github.mfvanek.pg.health.checks.cluster.ForeignKeysNotCoveredWithIndexCheckOnCluster;
import io.github.mfvanek.pg.health.checks.common.DatabaseCheckOnCluster;
import io.github.mfvanek.pg.model.constraint.ForeignKey;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration(proxyBeanMethods = false)
public class MigrationGeneratorConfig {

    @Bean
    public DbMigrationGenerator<ForeignKey> dbMigrationGenerator() {
        return new ForeignKeyMigrationGenerator(GeneratingOptions.builder().build());
    }

    @Bean
    public DatabaseCheckOnCluster<ForeignKey> foreignKeysNotCoveredWithIndex(final HighAvailabilityPgConnection haPgConnection) {
        return new ForeignKeysNotCoveredWithIndexCheckOnCluster(haPgConnection);
    }
}

File: pg-index-health-spring-boot-kotlin-demo/src/main/kotlin/io/github/mfvanek/pg/index/health/demo/kotlin/config/MigrationGeneratorConfig.kt
/*
 * Copyright (c) 2019-2025. Ivan Vakhrushev and others.
 * https://github.com/mfvanek/pg-index-health-demo
 *
 * Licensed under the Apache License 2.0
 */

package io.github.mfvanek.pg.index.health.demo.kotlin.config

import io.github.mfvanek.pg.connection.HighAvailabilityPgConnection
import io.github.mfvanek.pg.generator.DbMigrationGenerator
import io.github.mfvanek.pg.generator.ForeignKeyMigrationGenerator
import io.github.mfvanek.pg.generator.GeneratingOptions
import io.github.mfvanek.pg.health.checks.cluster.ForeignKeysNotCoveredWithIndexCheckOnCluster
import io.github.mfvanek.pg.health.checks.common.DatabaseCheckOnCluster
import io.github.mfvanek.pg.model.constraint.ForeignKey
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration(proxyBeanMethods = false)
class MigrationGeneratorConfig {

    @Bean
    fun dbMigrationGenerator(): DbMigrationGenerator<ForeignKey> {
        return ForeignKeyMigrationGenerator(GeneratingOptions.builder().build())
    }

    @Bean
    fun foreignKeysNotCoveredWithIndex(
        haPgConnection: HighAvailabilityPgConnection
    ): DatabaseCheckOnCluster<ForeignKey> {
        return ForeignKeysNotCoveredWithIndexCheckOnCluster(haPgConnection)
    }
}
